src/test/scala/org/constellation/domain/redownload/RedownloadServiceTest.scala

```diff
@@ -1,72 +1,231 @@
-
-      val persistFirst = redownloadService.persistOwnSnapshot(2L, "aaaa")
-      val persistSecond = redownloadService.persistOwnSnapshot(4L, "bbbb")
+      val redownloadService = RedownloadService[IO](dao)
+      val firstSnapshot = RecentSnapshot("aaaa", 2L, Map.empty)
+      val secondSnapshot = RecentSnapshot("bbbb", 4L, Map.empty)
+      val persistFirst = redownloadService.persistOwnSnapshot(2L, firstSnapshot)
+      val persistSecond = redownloadService.persistOwnSnapshot(4L, secondSnapshot)
       val check = redownloadService.getOwnSnapshots()
 
-      (persistFirst >> persistSecond >> check).unsafeRunSync shouldBe Map(2L -> "aaaa", 4L -> "bbbb")
+      (persistFirst >> persistSecond >> check).unsafeRunSync shouldBe Map(2L -> RecentSnapshot("aaaa", 2L, Map.empty),
+                                                                          4L -> RecentSnapshot("bbbb", 4L, Map.empty))
     }
   }
 
   "getOwnSnapshot" - {
     "should return hash if snapshot at given height exists" in {
-      val redownloadService = RedownloadService[IO]()
+      val redownloadService = RedownloadService[IO](dao)
+      val newSnapshot = RecentSnapshot("aaaa", 2L, Map.empty)
 
-      val persist = redownloadService.persistOwnSnapshot(2L, "aaaa")
+      val persist = redownloadService.persistOwnSnapshot(2L, newSnapshot)
       val check = redownloadService.getOwnSnapshot(2L)
 
-      (persist >> check).unsafeRunSync shouldBe "aaaa".some
+      (persist >> check).unsafeRunSync shouldBe RecentSnapshot("aaaa", 2L, Map.empty).some
     }
 
     "should return None if snapshot at given height does not exist" in {
-      val redownloadService = RedownloadService[IO]()
+      val redownloadService = RedownloadService[IO](dao)
 
       val check = redownloadService.getOwnSnapshot(2L)
 
       check.unsafeRunSync shouldBe none[String]
     }
   }
 
+  "fetchPeersProposals" - {
+    "should update peersProposals" in {
+      val redownloadService = RedownloadService[IO](dao)
+      val updateNewProps = redownloadService.updatePeerProps(proposals)
+
+      updateNewProps.unsafeRunSync().values.forall(_.size == numFacilitators) shouldBe true
+    }
+
+    "should not update peersProposals if a new proposal at the same height as an old proposal is recieved" in {
+      val redownloadService = RedownloadService[IO](dao)
+      val invalidProposdals = proposals :+ (proposals.head._1, RecentSnapshot("invalidProposdals", 0L, Map.empty))
+      val res = redownloadService.updatePeerProps(invalidProposdals).unsafeRunSync()
+
+      res.values.forall(_.size == numFacilitators) shouldBe true
+    }
+
+    "should not update peersProposals with a duplicate proposal" in {
+      val redownloadService = RedownloadService[IO](dao)
+      val invalidProposdals = proposals :+ proposals.head
+      val res = redownloadService.updatePeerProps(invalidProposdals).unsafeRunSync()
+      val check = res.values.forall(_.size == numFacilitators)
+      check shouldBe true
+    }
+  }
```

Comment:
There are only tests to check if `updatePeerProps` works correctly, you're not checking `fetchPeersProposals` at all :(